### PR TITLE
Azure File: optional CSI workflow and unit test fix

### PR DESCRIPTION
**Pull request by bertinatto**
CSI workflow is option for Azure File Driver, Operator and CSO repos. The reason is [a test that is consistently failing](https://bugzilla.redhat.com/show_bug.cgi?id=2020293).

Also, skip "vet" during testing for now. Even though all unit tests are passing, "go test" would return a non-zero return code. The reason was "vet" (called by "go test") was being OOM-killed.

CC @openshift/storage

### CODE REVIEW

File: ci-operator/config/openshift/azure-file-csi-driver-operator/openshift-azure-file-csi-driver-operator-main.yaml [hunk 1]
```diff
@@ -50,12 +50,7 @@ tests:
   commands: make verify
   container:
     from: src
-- as: e2e-azure
```
Comment (bertinatto) [inline ci-operator/config/openshift/azure-file-csi-driver-operator/openshift-azure-file-csi-driver-operator-main.yaml:53]: Azure File operator won't be installed here (it requires the feature gate), so there's no need to schedule this job.

**Comment by jsafrane** [comment]
I can see that at least snapshot tests are consistently failing. Do we really want to make the test mandatory?

**Comment by jsafrane** [comment]
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/23165/rehearse-23165-pull-ci-openshift-azure-file-csi-driver-operator-release-4.10-e2e-azure-file-csi/1455239159387000832

: External Storage [Driver: file.csi.azure.com] [Testpattern: Dynamic PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [LinuxOnly] expand_more15m0s | : External Storage [Driver: file.csi.azure.com]
[Testpattern: Dynamic PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [LinuxOnly]
: External Storage [Driver: file.csi.azure.com] [Testpattern: Dynamic PV (default fs)] subPath should be able to unmount after the subpath directory is deleted [LinuxOnly]
: External Storage [Driver: file.csi.azure.com] [Testpattern: Pre-provisioned Snapshot (delete policy)] snapshottable[Feature:VolumeSnapshotDataSource] volume snapshot controller should check snapshot fields, check restore correctly works after modifying source data, check deletion
: External Storage [Driver: file.csi.azure.com] [Testpattern: Pre-provisioned Snapshot (delete policy)] snapshottable[Feature:VolumeSnapshotDataSource] volume snapshot controller should check snapshot fields, check restore correctly works after modifying source data, check deletion
: External Storage [Driver: file.csi.azure.com] [Testpattern: Pre-provisioned Snapshot (delete policy)] snapshottable[Feature:VolumeSnapshotDataSource] volume snapshot controller should check snapshot fields, check restore correctly works after modifying source data, check deletion
: External Storage [Driver: file.csi.azure.com] [Testpattern: Pre-provisioned Snapshot (retain policy)] snapshottable[Feature:VolumeSnapshotDataSource] volume snapshot controller should check snapshot fields, check restore correctly works after modifying source data, check deletion

**Comment by bertinatto** [comment]
> I can see that at least snapshot tests are consistently failing. Do we really want to make the test mandatory?

@jsafrane my idea was to fix (or temporarily disable) the snapshot tests [here](https://bugzilla.redhat.com/show_bug.cgi?id=2019128) and keep the job mandatory for catching potential flaking tests. I can make them optional if you prefer.

**Comment by jsafrane** [comment]
Looks much better, still `subPath should be able to unmount after the subpath directory is deleted` fails quite consistently.

**Comment by bertinatto** [comment]
> Looks much better, still `subPath should be able to unmount after the subpath directory is deleted` fails quite consistently.

It's a known issue: https://github.com/kubernetes-sigs/azurefile-csi-driver/issues/668

**Comment by jsafrane** [comment]
Should we skip the test as known to be broken then? Otherwise we won't be able to merge anything with this PR.

**Comment by bertinatto** [comment]
/test all
Making sure unit tests won't flake

**Review by tsmetana — APPROVED** [review]
/lgtm

**Pull request merged**